### DISCUSSION TITLE
Why does talos rely on a separate config from kubernetes?

### DISCUSSION THREAD

**Discussion by wokalski** [siderolabs/talos#5289] [Q&A] [answered]
I don't understand why there are two configs; one for talos and then your regular kubernetes setup. Why isn't everything defined in terms of kubernetes resources?

**Comment by smira** [comment, marked as answer]
I'm not sure what kind of config you're asking about. If that's `talosconfig` and `kubeconfig` (client-side configuration), then the answer is easy. The file `talosconfig` file contains client credentials to access Talos Linux API. Sometimes Kubernetes might be down for a number of reasons (`etcd` issues, misconfiguration, etc.), while Talos API access will always be available. Talos API is a way to access the operating system and fix issues, e.g. fixing access to Kubernetes. When the OS is running fine, using only Kubernetes APIs (via `kubeconfig`) is all you might need to deploy and manage Kubernetes workloads.

**Reply by wokalski** [reply]
That's exactly what I meant. Thank you.
